cardano-node/src/Cardano/Node/Protocol/Byron.hs

```diff
@@ -0,0 +1,159 @@
+{-# LANGUAGE NamedFieldPuns #-}
+
+module Cardano.Node.Protocol.Byron
+  (
+    -- * Protocol exposing the specific type
+    -- | Use this when you need the specific instance
+    mkConsensusProtocolByron
+
+    -- * Protocols hiding the specific type
+    -- | Use this when you want to handle protocols generically
+  , mkSomeConsensusProtocolByron
+    -- * Errors
+  , ByronProtocolInstantiationError(..)
+  , renderByronProtocolInstantiationError
+
+    -- * Reusable parts
+  , readGenesis
+  , readLeaderCredentials
+  ) where
+
+
+import           Cardano.Prelude
+
+import           Codec.CBOR.Read (DeserialiseFailure, deserialiseFromBytes)
+import           Control.Monad.Trans.Except (ExceptT)
+import           Control.Monad.Trans.Except.Extra (bimapExceptT, firstExceptT,
+                                                   hoistEither, left)
+import qualified Data.ByteString.Lazy as LB
+
+import           Cardano.Api.Protocol.Byron (ByronProtocolInstantiationError(..),
+                   renderByronProtocolInstantiationError)
+import qualified Cardano.Chain.Genesis as Genesis
+import qualified Cardano.Chain.Update as Update
+import qualified Cardano.Chain.UTxO as UTxO
+import qualified Cardano.Crypto.Signing as Signing
+import           Cardano.Crypto.ProtocolMagic (RequiresNetworkMagic)
+
+import           Ouroboros.Consensus.Cardano hiding (Protocol)
+import qualified Ouroboros.Consensus.Cardano as Consensus
+
+import           Ouroboros.Consensus.Byron.Ledger (ByronBlock)
+
+import           Cardano.Config.Types
+                   (NodeByronProtocolConfiguration (..),
+                    ProtocolFilepaths(..), GenesisFile (..),
+                    SomeConsensusProtocol(..))
+import           Cardano.TracingOrphanInstances.Byron ()
+
+------------------------------------------------------------------------------
+-- Real Byron protocol
```

Comment:
We can drop the "real" now that we call it the Byron protocol, rather than before when we called it "RealPBFT".